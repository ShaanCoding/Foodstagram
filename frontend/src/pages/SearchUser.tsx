// import styles from "../styles/Search.module.css";
import { useParams } from 'react-router-dom'

import UseSearchUserMutation from '../api/UseSearchUserMutation';
import useAuth from '../api/util/useAuth'

const SearchUser = () => {
  const [account, isLoading] = useAuth()
  const { searchStr } = useParams();
  const searchUserMutation = UseSearchUserMutation()

  if (searchStr !== undefined)
    searchUserMutation.mutate({
      searchStr: searchStr,
    })
  
  console.log(searchStr)
  console.log(searchUserMutation.data?.data.data)

  return (
    <>
      <div>
        <h1 className='text-justify text-2xl px-48'>Search results for '{searchStr}'</h1>
        {searchUserMutation.data?.data.data !== undefined}
      </div>
    </>
  )
}

export default SearchUser