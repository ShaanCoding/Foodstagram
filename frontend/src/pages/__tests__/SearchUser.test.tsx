import { QueryClient, QueryClientProvider } from 'react-query'
import { BrowserRouter, Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import SearchUser from '../SearchUser'

const queryClient = new QueryClient()

it('The Search User/Profile page renders correctly', () => {
	const tree = renderer
		.create(
			<QueryClientProvider client={queryClient}>
				<BrowserRouter>
					<SearchUser />
				</BrowserRouter>
			</QueryClientProvider>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
