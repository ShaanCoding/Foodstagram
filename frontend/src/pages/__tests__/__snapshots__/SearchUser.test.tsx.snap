// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Search User/Profile page renders correctly 1`] = `
<div>
  <h1
    className="text-justify text-2xl px-48"
  >
    Search results for '
    '
  </h1>
</div>
`;
